src/kixi/collect/campaign/aggregate.clj

```diff
@@ -0,0 +1,66 @@
+(ns kixi.collect.campaign.aggregate
+  (:require [clojure.spec.alpha :as s]
+            [kixi.spec :refer [alias]]
+            [kixi.spec.conformers :as sc]
+            [kixi.comms.time :as t]
+            [kixi.collect.definitions]
+            [kixi.collect.datastore :as datastore]
+            [kixi.collect.aggregate :as agr]))
+
+(alias 'event 'kixi.event)
+(alias 'cr 'kixi.collect.request)
+(alias 'cc 'kixi.collect.campaign)
+(alias 'ms 'kixi.datastore.metadatastore)
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+
+(defprotocol ICampaignAggregate
+  (get-by-id [this id]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; EVENTS
+
+(s/fdef process-collection-requested-event
+        :args (s/cat :event (s/and :kixi/event
+                                   #(= [:kixi.collect/collection-requested "1.0.0"]
+                                       ((juxt ::event/type ::event/version) %))))
```

Comment:
we do this check quite often across our code bases. I wonder if it should live somewhere.